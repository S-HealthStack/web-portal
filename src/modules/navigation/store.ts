import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';

export enum Path {
  Root = '/',

  SignIn = '/signin',
  AccountActivation = '/account/activation',
  AccountCreate = '/account/create',
  AccountConfirm = '/account/confirm',
  AccountVerification = '/email-verification',
  ForgotPassword = '/account/forgot',
  ForgotPasswordConfirm = '/account/forgot/confirm',
  ResetPassword = '/account/forgot/reset',
  ResetPasswordComplete = '/account/forgot/complete',

  CreateStudy = '/create-study',

  Overview = '/overview',
  OverviewSubject = '/overview/participants/:subjectId',
  TrialManagement = '/trial-management',
  TrialManagementSubject = '/trial-management/participants/:subjectId',
  TrialManagementEditSurvey = '/trial-management/surveys/:surveyId/edit',
  TrialManagementSurveyResults = '/trial-management/surveys/:surveyId/results',
  UserAnalytics = '/user-analytics',
  DataCollection = '/data-collection',
  DataCollectionSubject = '/data-collection/participants/:subjectId',
  StudySettings = '/study-settings',
}

export const makeHistory = () =>
  process.env.NODE_ENV !== 'test'
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.OverviewSubject,
  Path.TrialManagement,
  Path.UserAnalytics,
  Path.DataCollection,
  Path.StudySettings,
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
