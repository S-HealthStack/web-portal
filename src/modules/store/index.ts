export type { RootState, AppDispatch, AppThunk } from './store';
export { useAppDispatch, useAppSelector } from './hooks';
export * from './tools';
