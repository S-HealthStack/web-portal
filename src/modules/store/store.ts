/* eslint-disable import/no-import-module-exports */
import {
  Action,
  ThunkAction,
  configureStore,
  getDefaultMiddleware,
  ThunkDispatch,
} from '@reduxjs/toolkit';
import { routerMiddleware } from 'connected-react-router';

import { history } from 'src/modules/navigation/store';
import rootReducer from 'src/modules/store/reducers';

export { rootReducer };

export const store = configureStore({
  reducer: rootReducer,
  middleware: [...getDefaultMiddleware(), routerMiddleware(history)],
  devTools: process.env.NODE_ENV !== 'production',
});

if (module.hot) {
  // Enable Webpack hot module replacement for reducers
  module.hot.accept('./reducers', async () => {
    // eslint-disable-next-line import/no-useless-path-segments
    const nextRootReducer = (await import('../store/reducers')).default;
    store.replaceReducer(nextRootReducer);
  });
}

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = ThunkDispatch<RootState, unknown, Action>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
