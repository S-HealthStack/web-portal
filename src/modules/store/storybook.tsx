import React from 'react';
import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { DecoratorFunction } from '@storybook/csf';
import { ReactFramework } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { rootReducer, RootState } from './store';

export const getStoreDecorator = (state: ConfigureStoreOptions<RootState>['preloadedState']) => {
  const store = configureStore({
    reducer: rootReducer,
    preloadedState: state,
  });

  return ((story) => (
    <Provider store={store}>
      <MemoryRouter>{story()}</MemoryRouter>
    </Provider>
  )) as DecoratorFunction<ReactFramework>;
};
