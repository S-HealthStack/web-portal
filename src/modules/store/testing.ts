import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { rootReducer, RootState } from './store';

export const createTestStore = (state: ConfigureStoreOptions<RootState>['preloadedState']) =>
  configureStore({
    reducer: rootReducer(),
    preloadedState: state,
  });
