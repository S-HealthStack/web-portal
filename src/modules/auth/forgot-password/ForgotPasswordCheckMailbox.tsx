import React, { useCallback } from 'react';
import styled from 'styled-components';

import { colors, px, typography } from 'src/styles';
import EmailSuccessImg from 'src/assets/illustrations/email_success.svg';
import Link from 'src/common/components/Link';
import Card from 'src/common/components/Card';
import ResultMessage from 'src/modules/auth/common/ResultMessage';
import { SnackbarContainer } from 'src/modules/snackbar';
import { usePasswordRecovery } from 'src/modules/auth/forgot-password/forgotPassword.slice';
import useSearchParam from 'react-use/lib/useSearchParam';
import { useAppDispatch } from 'src/modules/store';
import { showSnackbar } from 'src/modules/snackbar/snackbar.slice';

const TestIdContainer = styled.div`
  height: 100%;
`;

const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
`;

const ContentCard = styled(Card)`
  width: 90%;
  max-width: ${px(800)};
  min-height: ${px(546)};
  > div {
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

const ResendEmailOffer = styled.div`
  ${typography.bodySmallRegular};
  margin-top: ${px(53)};
  a {
    ${typography.bodySmallSemibold};
    color: ${colors.textPrimaryBlue};
    display: inline-block;
    margin-left: ${px(8)};
  }
`;

const ForgotPasswordCheckMailbox = () => {
  const email = useSearchParam('email') || '';
  const dispatch = useAppDispatch();

  const { sendPasswordRecoveryRequest } = usePasswordRecovery();

  const handleResend = useCallback(
    async (evt: React.MouseEvent<HTMLAnchorElement>) => {
      evt.preventDefault();
      evt.stopPropagation();

      const isOk = await sendPasswordRecoveryRequest({ email });
      if (isOk) {
        dispatch(showSnackbar({ text: 'Email resent.' }));
      }
    },
    [dispatch, email, sendPasswordRecoveryRequest]
  );

  return (
    <TestIdContainer data-testid="forgot-password-check-mailbox">
      <Container>
        <ContentCard>
          <ResultMessage
            picture={<EmailSuccessImg />}
            title="Check Mailbox"
            compactTitleMargin
            description={
              <>
                We&apos;ve emailed a link to reset your password to
                <br />
                <strong>{email}</strong>
              </>
            }
          >
            <ResendEmailOffer>
              Didn&apos;t get the email?
              <Link
                to="/"
                onClick={handleResend}
                data-testid="forgot-password-check-mailbox-resend"
              >
                Resend the email
              </Link>
            </ResendEmailOffer>
          </ResultMessage>
        </ContentCard>
      </Container>
      <SnackbarContainer useSimpleGrid />
    </TestIdContainer>
  );
};

export default ForgotPasswordCheckMailbox;
