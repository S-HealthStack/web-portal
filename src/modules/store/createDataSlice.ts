import { useCallback, useEffect } from 'react';
import _isEqual from 'lodash/isEqual';
import useMount from 'react-use/lib/useMount';
import {
  ActionCreatorWithoutPayload,
  ActionCreatorWithPayload,
  createSlice,
  PayloadAction,
} from '@reduxjs/toolkit';
import type { Reducer } from 'redux';

import applyDefaultApiErrorHandlers from 'src/modules/api/applyDefaultApiErrorHandlers';
import { SnackbarInstance } from 'src/modules/snackbar/snackbar.slice';
import useShowSnackbar from 'src/modules/snackbar/useShowSnackbar';
import { AppThunk, ErrorType, RootState, WithLoadableData } from 'src/modules/store';

import { useAppDispatch, useAppSelector } from './hooks';

type PublicDataSliceState<D> = WithLoadableData<D>;

type EmptyFetchArgs = null;

export type DataSliceState<D, A> = PublicDataSliceState<D> & {
  fetchArgs: A | EmptyFetchArgs;
  prevFetchArgs: A | EmptyFetchArgs;
  isLoadingSilently?: boolean;
};

type DataSliceOptions<D, N extends string, A> = {
  name: N;
  fetchData: (args: A, dispatch: unknown) => Promise<D>;
};

type FetchActionOptions = {
  force?: boolean;
  silent?: boolean;
  onError?: (e?: unknown) => void;
};

type HookParams<A, D> = {
  initialData?: D;
  fetchArgs: A | boolean;
  refetchSilentlyOnMount?: boolean;
};

export type ShowOnErrorSnackbarParams = Omit<SnackbarInstance, 'id' | 'hideTimeoutId'> & {
  onAction?: () => void;
};

type DataSliceReturnType<D, N, A> = {
  name: N;
  reducer: Reducer<DataSliceState<D, A>>;
  stateSelector: (s: RootState) => DataSliceState<D, A>;
  hook: (
    params?: HookParams<A, D>,
    snackBarParams?: ShowOnErrorSnackbarParams
  ) => DataSliceState<D, A> & {
    fetch: (args: A, opts?: FetchActionOptions) => Promise<void>;
    refetch: () => Promise<void>;
    reset: () => void;
  };
  actions: {
    reset: ActionCreatorWithoutPayload;
    setData: ActionCreatorWithPayload<{ data: D }>;
    loadingStarted: ActionCreatorWithPayload<{ fetchArgs: A }>;
    loadingSuccess: ActionCreatorWithPayload<{ data: D }>;
    loadingFailed: ActionCreatorWithPayload<ErrorType>;
    fetch: (args: A, opts?: FetchActionOptions) => AppThunk<Promise<void>>;
    refetch: () => AppThunk<Promise<void>>;
  };
};

const isFetchArgsEmpty = <D, A>(
  fetchArgs: DataSliceState<D, A>['fetchArgs']
): fetchArgs is EmptyFetchArgs => fetchArgs === null;

function createDataSlice<D extends object, N extends string, A = void>(
  options: DataSliceOptions<D, N, A>
): DataSliceReturnType<D, N, A> {
  const initialState: DataSliceState<D, A> = {
    fetchArgs: null,
    prevFetchArgs: null,
  };

  const slice = createSlice({
    name: options.name,
    initialState,
    reducers: {
      setData(state, action: PayloadAction<{ data: D }>) {
        return Object.assign(state, {
          data: action.payload.data,
        });
      },
      loadingStarted(state, action: PayloadAction<{ fetchArgs: A; silent?: boolean }>) {
        return Object.assign(state, {
          isLoading: !action.payload.silent,
          isLoadingSilently: !!action.payload.silent,
          fetchArgs: action.payload.fetchArgs,
          prevFetchArgs: { ...state.fetchArgs },
          error: undefined,
        });
      },
      loadingSuccess(state, action: PayloadAction<{ data: D }>) {
        return Object.assign(state, {
          isLoading: false,
          isLoadingSilently: false,
          error: undefined,
          data: action.payload.data,
          prevFetchArgs: { ...state.fetchArgs },
        });
      },
      loadingFailed(state, action: PayloadAction<ErrorType>) {
        return Object.assign(state, {
          isLoading: false,
          isLoadingSilently: false,
          error: String(action.payload),
          data: undefined,
          prevFetchArgs: { ...state.fetchArgs },
        });
      },
      reset(state) {
        state.isLoading = undefined;
        state.isLoadingSilently = undefined;
        state.data = undefined;
        state.error = undefined;

        return Object.assign(state, initialState);
      },
    },
  });

  const sliceStateSelector = (state: RootState) =>
    state[options.name as keyof RootState] as DataSliceState<D, A>;

  const fetchThunk =
    (args: A, opts?: FetchActionOptions): AppThunk<Promise<void>> =>
    async (dispatch, getState) => {
      if (!opts?.force) {
        const { fetchArgs: currentArgs } = sliceStateSelector(getState());

        // fetch only if this is a first fetch or args have been changed
        if (!isFetchArgsEmpty(currentArgs) && _isEqual(currentArgs, args)) {
          return;
        }
      }
      dispatch(slice.actions.loadingStarted({ fetchArgs: args, silent: opts?.silent }));
      try {
        const data = await options.fetchData(args, dispatch);

        const { fetchArgs: currentArgs } = sliceStateSelector(getState());
        // discard received data
        // if another fetch with different args performed while fetchData was in progress
        if (_isEqual(args, currentArgs)) {
          dispatch(slice.actions.loadingSuccess({ data }));
        }
      } catch (err) {
        const { fetchArgs: currentArgs } = sliceStateSelector(getState());
        if (_isEqual(args, currentArgs)) {
          console.error(err);
          dispatch(
            slice.actions.loadingFailed(
              (err instanceof Error ? err.message : undefined) ||
                `Failed to fetch data ${options.name}`
            )
          );

          if (!applyDefaultApiErrorHandlers(err, dispatch, false)) {
            opts?.onError?.(err);
          }
        }
      }
    };

  const refetchThunk = (): AppThunk<Promise<void>> => async (dispatch, getState) => {
    const { fetchArgs } = sliceStateSelector(getState());
    if (!isFetchArgsEmpty(fetchArgs)) {
      await dispatch(fetchThunk(fetchArgs, { force: true }));
    }
  };

  const refetchSilentlyIfRequiredThunk =
    (): AppThunk<Promise<void>> => async (dispatch, getState) => {
      const state = sliceStateSelector(getState());
      if (!state.isLoading && !state.isLoadingSilently && !isFetchArgsEmpty(state.fetchArgs)) {
        await dispatch(fetchThunk(state.fetchArgs, { force: true, silent: !state.error }));
      }
    };

  const useSliceHook = (params?: HookParams<A, D>, snackBarParams?: ShowOnErrorSnackbarParams) => {
    const dispatch = useAppDispatch();
    const sliceState = useAppSelector((state) => sliceStateSelector(state));
    const isStudyListLoading = useAppSelector((state) => state.studies.isLoading);
    const showSnackBar = useShowSnackbar();

    const onError = () => {
      if (snackBarParams) {
        showSnackBar(snackBarParams);
      }
    };

    const hookResult = {
      fetch: useCallback(
        (args: A, opts?: FetchActionOptions) => dispatch(fetchThunk(args, { ...opts, onError })),
        // eslint-disable-next-line react-hooks/exhaustive-deps
        [dispatch]
      ),
      refetch: useCallback(() => dispatch(refetchThunk()), [dispatch]),
      reset: useCallback(() => dispatch(slice.actions.reset()), [dispatch]),
      ...sliceState,
      isLoading: sliceState.isLoading || isStudyListLoading,
    };

    const hasParams = !!params;
    useEffect(() => {
      if (hasParams && typeof params.fetchArgs !== 'boolean') {
        dispatch(fetchThunk(params.fetchArgs, { onError }));
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [dispatch, hasParams, params?.fetchArgs]);

    useEffect(() => {
      if (
        !params?.initialData ||
        (!isFetchArgsEmpty(sliceState.fetchArgs) &&
          _isEqual(sliceState.fetchArgs, params?.fetchArgs))
      ) {
        return;
      }

      dispatch(slice.actions.setData({ data: params.initialData }));
    }, [dispatch, params?.fetchArgs, params?.initialData, sliceState.data, sliceState.fetchArgs]);

    useMount(() => {
      if (params?.refetchSilentlyOnMount) {
        dispatch(refetchSilentlyIfRequiredThunk());
      }
    });

    return hookResult;
  };

  return {
    name: options.name,
    reducer: slice.reducer,
    stateSelector: sliceStateSelector,
    hook: useSliceHook,
    actions: {
      ...slice.actions,
      fetch: fetchThunk,
      refetch: refetchThunk,
    },
  };
}

export default createDataSlice;
