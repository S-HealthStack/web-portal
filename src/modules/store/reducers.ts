import { combineReducers } from '@reduxjs/toolkit';
import { connectRouter } from 'connected-react-router';
import type { Reducer } from 'redux';

import authReducer from 'src/modules/auth/auth.slice';
import studySettingsReducer from 'src/modules/study-settings/studySettings.slice';
import snackbarReducer from 'src/modules/snackbar/snackbar.slice';
import overviewReducers from 'src/modules/overview/overview.slice';
import studiesReducer from 'src/modules/studies/studies.slice';
import studyManagementReducer from 'src/modules/study-management/studyManagement.slice';
import dataCollectionReducer from 'src/modules/data-collection/dataCollection.slice';
import sidebarReducer from 'src/modules/main-layout/sidebar/sidebar.slice';
import fileUploadReducer from 'src/modules/file-upload/fileUpload.slice';
import { history } from 'src/modules/navigation/store';
import { DataSliceState } from 'src/modules/store/createDataSlice';
import forgotPasswordReducers from 'src/modules/auth/forgot-password/forgotPassword.slice';
import resetPasswordReducers from 'src/modules/auth/forgot-password/resetPassword.slice';
import objectStorageReducers from 'src/modules/object-storage/objectStorage.slice';

const createReducer = <T, V>(asyncReducers?: { [key: string]: Reducer<DataSliceState<T, V>> }) =>
  combineReducers({
    ...authReducer,
    router: connectRouter(history),
    snackbar: snackbarReducer,
    dataCollection: dataCollectionReducer,
    sidebar: sidebarReducer,
    ...studySettingsReducer,
    ...overviewReducers,
    ...studyManagementReducer,
    ...forgotPasswordReducers,
    ...resetPasswordReducers,
    ...fileUploadReducer,
    ...asyncReducers,
    ...objectStorageReducers,
    ...studiesReducer,
  });

export default createReducer;
