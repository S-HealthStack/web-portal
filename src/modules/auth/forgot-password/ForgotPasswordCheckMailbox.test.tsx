import React from 'react';
import 'jest-styled-components';
import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import theme from 'src/styles/theme';
import { makeHistory } from 'src/modules/navigation/store';
import { ThemeProvider } from 'styled-components/';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { makeStore } from 'src/modules/store/store';
import ForgotPasswordCheckMailbox from 'src/modules/auth/forgot-password/ForgotPasswordCheckMailbox';

describe('ForgotPasswordCheckMailbox', () => {
  let store: ReturnType<typeof makeStore>;
  let history: ReturnType<typeof makeHistory>;

  beforeEach(() => {
    history = makeHistory();
    store = makeStore(history);
  });

  let search = '';
  const mockSearch = (s: string) => {
    search = s;
    Object.defineProperty(window, 'location', {
      configurable: true,
      enumerable: true,
      get: () => new URL(`http://test.com/?${search}`),
    });
  };

  it('should render', async () => {
    const email = 'hello@samsung.com';
    mockSearch(`email=${email}`);

    await act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <ForgotPasswordCheckMailbox />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    expect(await screen.findByTestId('forgot-password-check-mailbox')).toMatchSnapshot();

    const resendButton = await screen.findByTestId('forgot-password-check-mailbox-resend');

    await userEvent.click(resendButton);
  });

  it('[NEGATIVE] should render without email', async () => {
    mockSearch(`email=`);

    const baseElement = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <ForgotPasswordCheckMailbox />
          </ConnectedRouter>
        </Provider>
      </ThemeProvider>
    );

    expect(baseElement).toBeDefined();
  });
});
