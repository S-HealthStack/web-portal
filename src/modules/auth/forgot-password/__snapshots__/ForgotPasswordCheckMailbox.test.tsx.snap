// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordCheckMailbox should render 1`] = `
.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: default;
  -webkit-user-drag: none;
}

.c10:hover {
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  box-sizing: border-box;
  padding: 1.5rem 1.5rem 1rem;
  overflow: visible;
  position: relative;
}

.c2 > * {
  min-width: 0;
  min-height: 0;
  overflow: visible;
}

.c4 {
  -webkit-transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  transition: opacity 0.3s cubic-bezier(0.45,0.05,0,1);
  opacity: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: -0.875rem;
}

.c5 > * {
  text-align: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 15rem;
  height: 15rem;
}

.c7 {
  font-family: Inter,sans-serif;
  font-size: 2rem;
  line-height: 2.625rem;
  font-weight: 600;
  font-style: normal;
  margin: 0;
  margin-top: 0.375rem;
}

.c8 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  max-width: 37.5rem;
  margin: 0;
  margin-top: 0.5rem;
}

.c8 strong {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c11 {
  position: relative;
  height: 0;
  width: 100%;
  pointer-events: none;
}

.c0 {
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.c3 {
  width: 90%;
  max-width: 50rem;
  min-height: 34.125rem;
}

.c3 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  margin-top: 3.3125rem;
}

.c9 a {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 600;
  font-style: normal;
  color: #4475E3;
  display: inline-block;
  margin-left: 0.5rem;
}

<div
  class="c0"
  data-testid="forgot-password-check-mailbox"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c4"
        data-testid="card-content"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <svg />
          </div>
          <h2
            class="c7"
          >
            Check Mailbox
          </h2>
          <p
            class="c8"
          >
            We've emailed a link to reset your password to
            <br />
            <strong>
              hello@samsung.com
            </strong>
          </p>
          <div
            class=""
          >
            <div
              class="c9"
            >
              Didn't get the email?
              <a
                class="c10"
                data-testid="forgot-password-check-mailbox-resend"
                href="/"
              >
                Resend the email
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c11"
  />
</div>
`;
